snippet_ranger/__main__.py

```diff
@@ -0,0 +1,106 @@
+import argparse
+import logging
+import sys
+
+from modelforge.logs import setup_logging
+from ast2vec.__main__ import one_arg_parser
+
+from snippet_ranger.model2.source2func import source2func_entry
+from snippet_ranger.dependent_reps import dependent_reps_entry, LibIO
+
+def get_parser() -> argparse.ArgumentParser:
+    """
+    Create main parser.
+
+    :return: Parser
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--log-level", default="INFO",
+                        choices=logging._nameToLevel,
+                        help="Logging verbosity.")
+
+    # Create all common arguments
+
+    model2input_arg = one_arg_parser(
+        "input", help="Directory to scan recursively for asdf files.")
+
+    process_arg = one_arg_parser(
+        "-p", "--processes", type=int, default=0,
+        help="Number of processes to use. 0 means CPU count.")
+
+    filter_arg = one_arg_parser(
+        "--filter", default="**/*.asdf", help="File name glob selector.")
+
+    library_name_arg = one_arg_parser(
+        "--library_name", help="Provide the name of the library.")
+
+    # Create and construct subparsers
+
+    subparsers = parser.add_subparsers(help="Commands", dest="command")
+
+    source2func_parser = subparsers.add_parser(
+        "source2func",
+        help="Decompose source model to functions were specified library is used. It makes model "
+             "entry from each function that uses library and produce one Function model from one "
```

Comment:
Entry means "the opposite of exit". It is pretty generic. Please avoid it in comments or names, it is used only when the fantasy segfaults.